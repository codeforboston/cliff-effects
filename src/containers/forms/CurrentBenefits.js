import { connect } from 'react-redux';

import { CurrentBenefitsStep } from '../../forms/CurrentBenefits';
import { setHasBenefit } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),

    USState: state.getIn([
      'geography',
      'state',
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setHasBenefit({ benefit, value }) {
      dispatch(
        setHasBenefit({
          time: 'current',
          benefit,
          value,
        })
      );
    },
  };
};

const CurrentBenefitsContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentBenefitsStep);

export { CurrentBenefitsContainer as CurrentBenefitsStep };
