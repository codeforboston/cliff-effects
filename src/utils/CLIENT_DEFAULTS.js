import { cloneDeep } from 'lodash';

/**
 * Default client values.
 * MUST ALWAYS BE VALID
 * 
 * Note: A 'positive number' money value can have more than
 *     two decimal places.
 */
const CLIENT_DEFAULTS = {

  // Property Name                | Default Value   | Valid Value Description

  current: {
    // Current programs
    hasSnap:                        false,            // boolean
    hasHousing:                     false,            // boolean
    // Household
    // 'm_' for 'member'
    household:  [{                  // 'head' member  // one or more member objects
        m_age:                      30,               // whole number
        m_role:                     'head',           // 'head', 'spouse', 'member'
        m_disabled:                 false             // boolean
    }],
    // Income
    earned:                         0,                // positive number (can have more than two decimal places)
    TAFDC:                          0,                // positive number
    SSI:                            0,                // positive number
    SSDI:                           0,                // positive number
    childSupportIn:                 0,                // positive number
    unemployment:                   0,                // positive number
    workersComp:                    0,                // positive number
    pension:                        0,                // positive number
    socialSecurity:                 0,                // positive number
    alimony:                        0,                // positive number
    otherIncome:                    0,                // positive number
    incomeExclusions:               0,                // positive number
    // Expenses
    childDirectCare:                0,                // positive number
    childBeforeAndAfterSchoolCare:  0,                // positive number
    childTransportation:            0,                // positive number
    childOtherCare:                 0,                // positive number
    earnedBecauseOfChildCare:       0,                // positive number
    childSupportPaidOut:            0,                // positive number
    adultDirectCare:                0,                // positive number
    adultTransportation:            0,                // positive number
    adultOtherCare:                 0,                // positive number
    disabledAssistance:             0,                // positive number
    earnedBecauseOfAdultCare:       0,                // positive number
    disabledMedical:                0,                // positive number
    otherMedical:                   0,                // positive number
    /** @todo When client has section 8, switch this to 'housingVoucher' */
    /** @todo Make shelter expense values more robust. */
    shelter:                        'homeless',       // 'homeless', 'homeowner', 'renter', 'housingVoucher'
    contractRent:                   0,                // positive number
    rentShare:                      0,                // positive number
    rent:                           0,                // positive number
    mortgage:                       0,                // positive number
    housingInsurance:               0,                // positive number
    propertyTax:                    0,                // positive number
    climateControl:                 false,            // boolean
    nonHeatElectricity:             false,            // boolean
    phone:                          false,            // boolean
    fuelAssistance:                 false,            // boolean  Note: Input is 'yes', 'no', but should be being converted
    otherExpenses:                  0                 // positive number
  },

  // Note: A `future` value will default to the client's `current` value until a user changes it explicitly
  future: {},                       // same structure // same

};

CLIENT_DEFAULTS.future = cloneDeep( CLIENT_DEFAULTS.current );


export { CLIENT_DEFAULTS };
