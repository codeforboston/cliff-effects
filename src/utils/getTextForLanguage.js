/** Returns a translator based on the language given */

// TRANSFORMER FUNCTIONS
import { mergeWith } from 'lodash';
import { interpolateSnippets } from './interpolation.js';

// DATA
import { getLocalizationData } from '../localization/all';
import inlineComponents from '../localization/inlineComponents';

// Get copy of localization data
const localizations = getLocalizationData();

// store interpolated and (if necessary) merged snippets objects
let finishedSnippets = { en: interpolateSnippets(localizations.en, inlineComponents) };

/** Customizes Lodash's mergeWith function to replace arrays completely
 * (to avoid arrays of English strings being mixed with arrays of translated
 * strings, if they happen to have different lengths).
 */
const mergeCustomizer = function (objValue, srcValue) {
  if (Array.isArray(objValue)) {
    return srcValue;
  }
};

/** Returns the object named by langName that contains
 * the text snippets of that language. If that language
 * doesn't exist, it warns the coder and returns English.
 */
const getTextForLanguage = function (langCode) {
  if (!localizations[ langCode ]) {
    console.warn('There\'s no localization for ' + langCode + '. Defaulting to English.');
    langCode = 'en';
  }

  if (!finishedSnippets[ langCode ]) {
    // interpolate snippets and merge with English (filling in any gaps)
    const interpolatedSnippets = interpolateSnippets(localizations[ langCode ], inlineComponents);

    // deeply merge the object containing snippets in langName with English,
    // so that we fall back to English if a particular field is missing.
    finishedSnippets[ langCode ] =  mergeWith({}, finishedSnippets.en, interpolatedSnippets, mergeCustomizer);
  }

  return finishedSnippets[ langCode ];
};  // End getTextForLanguage()


export { getTextForLanguage };
