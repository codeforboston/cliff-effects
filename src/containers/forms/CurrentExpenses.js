import { connect } from 'react-redux';

import { CurrentExpensesStep } from '../../forms/CurrentExpenses';

import { setCashValue, setHousingType, setPaysUtility, setGetsFuelAssistance } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setExpenseValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },

    setHousingType({ housingType }) {
      dispatch(
        setHousingType({
          time: 'current',
          housingType,
        })
      );
    },

    setPaysUtility({ utility, paysUtility }) {
      dispatch(
        setPaysUtility({
          time: 'current',
          utility,
          paysUtility,
        })
      );
    },

    setGetsFuelAssistance({ getsAssistance }) {
      dispatch(
        setGetsFuelAssistance({
          time: 'current',
          getsAssistance,
        })
      );
    },
  };
};

const CurrentExpensesStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentExpensesStep);

export { CurrentExpensesStepContainer as CurrentExpensesStep };
