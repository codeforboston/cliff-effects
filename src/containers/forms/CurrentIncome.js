import { connect } from 'react-redux';

import { CurrentIncomeStep } from '../../forms/CurrentIncome';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setIncomeValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },
  };
};

const CurrentIncomeStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentIncomeStep);

export { CurrentIncomeStepContainer as CurrentIncomeStep };
