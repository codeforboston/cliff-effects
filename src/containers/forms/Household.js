import { connect } from 'react-redux';

import { HouseholdStep } from '../../forms/Household';
import {
  addMember,
  removeMember,
  setMemberIsDisabled,
  setMemberRole,
  setMemberAge,
} from '../../actions';

const time = 'current';

const mapStateToProps = (state) => {
  return {
    household: state.getIn([
      'client',
      'current',
      'household', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setMemberAge({ index, age }) {
      dispatch(
        setMemberAge({
          time,
          index,
          age,
        })
      );
    },

    setMemberIsDisabled({ index, isDisabled }) {
      dispatch(
        setMemberIsDisabled({
          time,
          index,
          isDisabled,
        })
      );
    },

    setMemberRole({ index, role }) {
      dispatch(
        setMemberRole({
          time,
          index,
          role,
        })
      );
    },

    removeMember({ index }) {
      dispatch(
        removeMember({
          time,
          index,
        })
      );
    },

    addMember({ member }) {
      dispatch(
        addMember({
          time,
          member,
        })
      );
    },
  };
};

const HouseholdStepContainer = connect(mapStateToProps, mapDispatchToProps)(HouseholdStep);

export { HouseholdStepContainer as HouseholdStep };
