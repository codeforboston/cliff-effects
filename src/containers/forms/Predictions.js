import { connect } from 'react-redux';

import { PredictionsStep } from '../../forms/Predictions';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return { client: state.get('client') };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setPredictionValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'future',
          name,
          value,
        })
      );
    },
  };
};

const PredictionsStepContainer = connect(mapStateToProps, mapDispatchToProps)(PredictionsStep);

export { PredictionsStepContainer as PredictionsStep };
