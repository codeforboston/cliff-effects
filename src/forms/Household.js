// REACT COMPONENTS
import React from 'react';
import {
  Button,
  Form,
  Dropdown,
  Header,
  Checkbox,
  Icon,
} from 'semantic-ui-react';

// PROJECT COMPONENTS
import {
  FormPartsContainer,
  ColumnHeading,
  ManagedNumberField,
} from './formHelpers';

// COMPONENT HELPER FUNCTIONS
import { getTimeSetter } from '../utils/getTimeSetter';
import { isPositiveWholeNumber } from '../utils/validators';

// OBJECT MANIPULATION
import { cloneDeep } from 'lodash';


// ======================
// GENERICS
// ======================
// To be able to adjust sizes easily
// Very specific to household size. May be worth creating
// a constructor for columns in general, or maybe use a Grid.
const columnStyle = {
  display:   'inline-block',
  textAlign: 'center',
  marginTop: '0.7em',
  // marginBottom: '0.7em'
};

const Columns = {};

// `noMargin` is a bit hacky, but it'll do for now
Columns.One = function ({ noMargin, children }) {
  var marginTop = columnStyle.marginTop;
  if (noMargin) { 
    marginTop = 0; 
  }
  return (<div style={{ ...columnStyle, marginTop: marginTop, width: '5em' }}> {children} </div>);
};

Columns.Two = function ({ noMargin, children }) {
  var marginTop = columnStyle.marginTop;
  if (noMargin) { 
    marginTop = 0; 
  }
  return (<div style={{ ...columnStyle, marginTop: marginTop, width: '20em', textAlign: 'left', paddingLeft: '1em' }}> {children} </div>);
};

Columns.Three = function ({ noMargin, children }) {
  var marginTop = columnStyle.marginTop;
  if (noMargin) { 
    marginTop = 0; 
  }
  return (<div style={{ ...columnStyle, marginTop: marginTop, width: '5em' }}> {children} </div>);
};

Columns.Four = function ({ noMargin, children }) {
  var marginTop = columnStyle.marginTop;
  if (noMargin) {
    marginTop = 0;
  }
  return (<div style={{ ...columnStyle, marginTop: marginTop, width: '10em' }}> {children} </div>);
};


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const ColumnHeader = function ({ children, columnNum }) {

  var Container = Columns[ columnNum ];

  return (
    <Container>
      <ColumnHeading
        type={ 'household' }
        colName={ '' }>
        { children }
      </ColumnHeading>
    </Container>
  );

};


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const MemberButton = function ({ basic, color, iconName, className, onClick }) {

  color = color || null;

  return (
    <Button
      type={ 'button' }
      basic={ !!basic }
      color={ color }
      icon={ iconName }
      className={ className }
      onClick={ onClick }
      style={{ padding: '0', height: '2.2em', width: '2.2em' }}
      circular />
  );

};


// ======================
// UNIQUE
// ======================

/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const Role = function ({ member, setMember }) {

  var ThisRole  = null,
      margin   = '0';

  if (member.index === 0) {

    ThisRole  = <span>Head of Household</span>;

  } else if (member.index === 1) {

    margin = '-1em';

    var options = [ { text: 'Spouse of Head of Household', value: 'spouse' }, { text: 'Child/Other Household Member', value: 'member' } ];

    ThisRole = <Dropdown
      selection
      name={ 'm_role' }
      value={ member.m_role }
      options={ options }
      onChange={ setMember } />;

  } else {

    ThisRole = <span>Child/Other Household Member</span>;

  }

  // Styles will have to be adjusted.
  return (
    <div style={{ display: 'inline-block', width: '100%', textAlign: 'left', marginLeft: margin }}>
      { ThisRole }
    </div>
  );

};  // End Role(<>)


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const MemberField = function ({ household, time, setHousehold, setClientProperty }, indx) {

  var member      = household[ indx ],
      routeStart  = 'household/' + indx + '/';
  member.index    = indx;  // Just needed as member prop in this file


  var onMemberChange = function (evnt, inputProps) {
    var route = routeStart + inputProps.name;
    var data  = { route: route, value: inputProps.value };
    setClientProperty(evnt, data);
  };


  var onMemberChecked = function (evnt, inputProps) {
    var route = routeStart + inputProps.name;
    var data  = { route: route, value: inputProps.checked };
    setClientProperty(evnt, data);
  };


  var removeMember = function (evnt, inputProps) {
    household.splice(indx, 1);
    setHousehold(evnt, household);
  };  // End removeMember()


  // The font size thing is a bit weird, but... later
  return (
    <Form.Field
      className='flex-item'
      key={ indx }>

      <Columns.One>
        { indx > 0
          ? <MemberButton
            className={ 'remove' }
            onClick={ removeMember }
            iconName={ 'remove' } />
          : 
          <span>{ household.length > 1
            ? <Icon
              fitted
              name={ 'ban' }
              style={{ color: '#cfcfd0', fontSize: '2.2em', verticalAlign: 'text-top' }} />
            : null
          }
          </span>
        }
      </Columns.One>

      <Columns.Two>
        <Role
          member={ member }
          setMember={ onMemberChange } />
      </Columns.Two>

      <Columns.Three>
        <ManagedNumberField
          value      = { member.m_age }
          name       = { 'm_age' }
          className  = { time + ' member-age ' + time }
          validation = { isPositiveWholeNumber }
          format     = { function (value) { return value; } }
          store      = { onMemberChange }
          onBlur     = { function () { return true; } } />
      </Columns.Three>

      <Columns.Four>
        <Checkbox
          name={ 'm_disabled' }
          checked={ member.m_disabled }
          onChange={ onMemberChecked } />
      </Columns.Four>

    </Form.Field>
  );

};  // End MemberField()


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const getMembers = function (current, time, setHousehold, setClientProperty) {

  var household = current.household,
      props     = {
        household:         household,
        time:              time,
        setHousehold:      setHousehold,
        setClientProperty: setClientProperty,
      };

  var mems = [];
  for (let memi = 0; memi < household.length; memi++) {
    mems.push(MemberField(props, memi));
  };

  return mems;

};  // End getMembers()


/** @todo description
*
* @todo Could this be a number field? If not, then a dropdown?
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const HouseholdContent = function ({ current, time, setClientProperty }) {

  // Don't mutate state properties
  var household = cloneDeep(current.household);


  var setHousehold = function (evnt, newHousehold) {

    var obj = {
      route: 'household',
      value: newHousehold,
    };

    setClientProperty(evnt, obj);

  };  // End setHousehold()


  var addMember = function (evnt, inputProps) {

    var member = household.length === 1
      ? { m_age: 30, m_role: 'spouse', m_disabled: false }
      : { m_age: 12, m_role: 'member', m_disabled: false };

    household.push(member);
    setHousehold(evnt, household);

  };  // End addMember()


  return (
    <div className='field-aligner two-column'>
      <div style={{ marginBottom: '.5em' }}>
        <ColumnHeader columnNum='One' />
        <ColumnHeader columnNum='Two'>Role</ColumnHeader>
        <ColumnHeader columnNum='Three'>Age</ColumnHeader>
        <ColumnHeader columnNum='Four'>Disabled</ColumnHeader>
      </div>

      { getMembers(current, time, setHousehold, setClientProperty) }

      <Button
        type={ 'button' }
        id={ 'addMember' }
        basic
        onClick={ addMember }>
        <Columns.One noMargin={ true }>
          <Icon
            name={ 'plus' }
            circular
            inverted
            color={ 'teal' } />
        </Columns.One>

        <Columns.Two noMargin={ true }>
          <Header
            as='h4'
            color={ 'teal' }> Add a member
          </Header>
        </Columns.Two>

        <Columns.Three noMargin={ true } />
        <Columns.Four noMargin={ true } />
      </Button>

    </div>
  );

};  // End HouseholdContent()


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
// `props` is a cloned version of the original props. References broken.
const HouseholdStep = function (props) {

  const setTimeProp = getTimeSetter('current', props.changeClient);

  return (
    <Form className='current-household-size-form flex-column flex-item'>
      <FormPartsContainer
        title     = { 'Household' }
        clarifier = { 'Information about the members of your household.' }
        left      = {{ name: 'Previous', func: props.previousStep }}
        right     = {{ name: 'Next', func: props.nextStep }}>
        <HouseholdContent
          setClientProperty={ setTimeProp }
          current={ props.client.current }
          time={ 'current' } />
      </FormPartsContainer>
    </Form>
  );

};  // End HouseholdStep()

export { HouseholdStep };
