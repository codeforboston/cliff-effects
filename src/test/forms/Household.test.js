import React from 'react';
import { shallow } from 'enzyme';

import { HouseholdStep } from '../../forms/Household';

import { CLIENT_DEFAULTS } from '../../utils/CLIENT_DEFAULTS';

// LOCALIZATION
import { getTextForLanguage } from '../../utils/getTextForLanguage';


var snippets = getTextForLanguage(`en`);
test('Household step component renders as snapshot correctly', () => {
  const navData = {
          left:   { text: 'Previous', onClick: jest.fn() },
          middle: null,
          right:  { text: 'Next', onClick: jest.fn() },
        },
        updateClientValue     = jest.fn(),
        saveForm         = jest.fn(),
        askToResetClient = jest.fn(),
        openFeedback     = jest.fn(),
        formSnippets     = snippets.visitPage.household;

  const wrapper = shallow(
    <HouseholdStep
      currentStep       = { 2 }
      client            = { CLIENT_DEFAULTS }
      navData           = { navData }
      updateClientValue      = { updateClientValue }
      saveForm          = { saveForm }
      askToResetClient  = { askToResetClient }
      openFeedback      = { openFeedback }
      snippets          = { formSnippets } />
  );

  expect(wrapper).toMatchSnapshot();
});
