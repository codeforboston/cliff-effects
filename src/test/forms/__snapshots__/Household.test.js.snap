// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Household step component renders as snapshot correctly 1`] = `
<Form
  as="form"
  className="current-household-size-form flex-column flex-item"
>
  <FormPartsContainer
    clarifier={
      <span
        className="snippet"
        lang="en"
      >
        Information about the members of your household.
      </span>
    }
    navData={
      Object {
        "left": Object {
          "onClick": [Function],
          "text": "Previous",
        },
        "middle": null,
        "right": Object {
          "onClick": [Function],
          "text": "Next",
        },
      }
    }
    title={
      <span
        className="snippet"
        lang="en"
      >
        Household
      </span>
    }
  >
    <HouseholdContent
      current={
        Object {
          "SSDI": 0,
          "SSI": 0,
          "TAFDC": 0,
          "adultDirectCare": 0,
          "adultOtherCare": 0,
          "adultTransportation": 0,
          "alimony": 0,
          "childBeforeAndAfterSchoolCare": 0,
          "childDirectCare": 0,
          "childOtherCare": 0,
          "childSupportIn": 0,
          "childSupportPaidOut": 0,
          "childTransportation": 0,
          "climateControl": false,
          "contractRent": 0,
          "disabledAssistance": 0,
          "disabledMedical": 0,
          "earned": 0,
          "earnedBecauseOfAdultCare": 0,
          "earnedBecauseOfChildCare": 0,
          "fuelAssistance": false,
          "hasSection8": false,
          "hasSnap": false,
          "household": Array [
            Object {
              "m_age": 30,
              "m_disabled": false,
              "m_role": "head",
            },
          ],
          "housing": "homeless",
          "housingInsurance": 0,
          "mortgage": 0,
          "nonHeatElectricity": false,
          "otherExpensesCable": 0,
          "otherExpensesCareProducts": 0,
          "otherExpensesClothes": 0,
          "otherExpensesEntertainment": 0,
          "otherExpensesFood": 0,
          "otherExpensesMedical": 0,
          "otherExpensesOther": 0,
          "otherExpensesPhone": 0,
          "otherExpensesTransport": 0,
          "otherExpensesUtilities": 0,
          "otherIncome": 0,
          "otherMedical": 0,
          "pension": 0,
          "phone": false,
          "propertyTax": 0,
          "rent": 0,
          "rentShare": 0,
          "socialSecurity": 0,
          "unemployment": 0,
          "wantsToSeeOtherExpenses": false,
          "workersComp": 0,
        }
      }
      snippets={
        Object {
          "i_addMember": <span
            className="snippet"
            lang="en"
        >
            Add a member
        </span>,
          "i_age": <span
            className="snippet"
            lang="en"
        >
            Age
        </span>,
          "i_childOther": <span
            className="snippet"
            lang="en"
        >
            Child/Other Household Member
        </span>,
          "i_clarifier": <span
            className="snippet"
            lang="en"
        >
            Information about the members of your household.
        </span>,
          "i_disabled": <span
            className="snippet"
            lang="en"
        >
            Disabled
        </span>,
          "i_headOfHousehold": <span
            className="snippet"
            lang="en"
        >
            Head of Household
        </span>,
          "i_next": <span
            className="snippet"
            lang="en"
        >
            Next
        </span>,
          "i_previous": <span
            className="snippet"
            lang="en"
        >
            Previous
        </span>,
          "i_role": <span
            className="snippet"
            lang="en"
        >
            Role
        </span>,
          "i_spouse": <span
            className="snippet"
            lang="en"
        >
            Spouse of Head of Household
        </span>,
          "i_title": <span
            className="snippet"
            lang="en"
        >
            Household
        </span>,
          "langCode": "en",
        }
      }
      time="current"
      updateClientValue={[Function]}
    />
  </FormPartsContainer>
</Form>
`;
